stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/TypingUpdatesBuffer.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.getstream.sdk.chat.utils.typing
+
+/**
+ * Designed to buffer typing inputs.
+ *
+ * Its implementation should receive keystroke events by calling [TypingUpdatesBuffer.onKeystroke]
+ * which it will internally buffer and send start and stop typing API calls accordingly.
+ * This cuts down on unnecessary API calls.
+ *
+ * For the default implementation see [DefaultTypingUpdatesBuffer].
+ */
+public interface TypingUpdatesBuffer {
+
+    /**
+     * Should be called on every input change.
+     */
+    public fun onKeystroke()
+
+    /**
+     * Should send a stop typing event manually.
+     *
+     * Useful for runtime hygiene such as responding to lifecycle events.
+     */
+    public fun clear()
```

Comment:
Called this clear and not `onStopTyping`  on purpose in order to signal that this is manually clearing the events, i.e. sending a stop typing event.

Actually tracking events and sending firing stop events after the delay interval has passed should be the job of the implementation class.

External stop typing events such as if the input is empty can often be unreliable because someone might have simply wanted to start over with their message, hence why there's an implicit signal towards the fact that these should be handled internall and we have an implementation mention above.